include/oneapi/mkl/rng/device/detail/poisson_impl.hpp

```diff
@@ -0,0 +1,355 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _MKL_RNG_DEVICE_POISSON_IMPL_HPP_
+#define _MKL_RNG_DEVICE_POISSON_IMPL_HPP_
+
+#include <limits>
+
+namespace oneapi::mkl::rng::device::detail {
+
+// Implementation of Poisson distribution uses 3 methods depending on lambda parameter:
+//    - table-lookup method [1] for small lambdas (lambda < 60)
+//    - Devroye's method [2] for medium lambdas (60 <= lambda < 1000)
+//    - Gaussian approximation [1] for huge lambdas (lambda >= 1000)
+//
+// References:
+// [1] Michael B. Giles // Algorithm 955: approximation of the inverse Poisson cumulative
```

Comment:
Just a minor: I suggest to align the way how we write references (here and in mrg_impl)

Reply:
Done